manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -282,6 +289,32 @@ func (tr *TaskReaper) tick() {
 			return nil
 		})
 	}
+
+	// Perform service cleanup
+	deleteServices := make(map[string]struct{})
+	// figure out which services need to be cleaned up
+	if len(tr.cleanupServices) > 0 {
+		tr.store.View(func(tx store.ReadTx) {
+			for svc := range tr.cleanupServices {
+				tasks, _ := store.FindTasks(tx, store.ByServiceID(svc))
+				if tasks != nil && len(tasks) == 0 {
```

Comment:
What if tasks == nil ?

Reply:
It seems like this doesn't happen from the scenarios I tested (when there are no tasks and the service is still around. I also looked at the code for `FindTasks` and it seems reasonable), but I'll double check.

Reply:
Yep, I guess tasks shouldn't be nil. Does it make sense to either remove the nil check, or add a condition where tasks ==  nil is also an indication to clean up ?

Reply:
Actually it seems like `len(tasks)` will be 0 even if `tasks` is `nil`. The right thing to do is probably to not discard the error.